pkg/controller/volume/persistentvolume/scheduler_binder.go

```diff
@@ -143,6 +143,9 @@ func (b *volumeBinder) GetBindingsCache() PodBindingCache {
 func (b *volumeBinder) FindPodVolumes(pod *v1.Pod, node *v1.Node) (unboundVolumesSatisfied, boundVolumesSatisfied bool, err error) {
 	podName := getPodName(pod)
 
+	// Pod binding cache may be out of date, clear for the given pod and node first.
+	b.podBindingCache.ClearBindings(pod, node.Name)
```

Comment:
Can this go after `getPodVolumes`, if number of total claims > 0?  That way we save a check with mutex for pods that don't use volumes.